feat: setup jsdoc for documentation

Samridhi-98 opened the pull request:
**Description**

- Added `conf.json`
- Modified `package.json`

**Resolves**
Fixes #17

Samridhi-98 commented:
@coditva  @elit-altum  please review my PR.

coditva commented on package.json:
```diff
@@ -12,7 +12,8 @@
     "dev:client": "next ./frontend",
-    "build": "node npm/build.js"
+    "build": "node npm/build.js",
+    "generate-docs": "jsdoc -c conf.json -d ./docs -r ./lib"
```
Let's keep the docs in the `out/` directory itself.

coditva commented on conf.json:
```diff
@@ -0,0 +1,26 @@
+{
```
Let's call this file something less generic like `jsdoc.conf.json`

Samridhi-98 replied:
Done.

coditva reviewed (changes requested):
@Samridhi-98 We might need to add JSDoc to the three sub-packages `dashboard`, `frontend` and `reporter`

Samridhi-98 commented:
> @Samridhi-98 We might need to add JSDoc to the three sub-packages `dashboard`, `frontend` and `reporter`

I have updated the `package.json script` for each subdirectory keeping `jsdoc.conf.json` the same for all.
Please review and let me know if I have implemented it correctly or not.

coditva commented on dashboard/package.json:
```diff
@@ -15,7 +15,8 @@
     "coverage:integration": "npm run test:integration && codecov",
-    "dev": "nodemon ./server/index.js"
+    "dev": "nodemon ./server/index.js",
+    "generate-docs": "jsdoc -c ../jsdoc.conf.json -d ../out/dashboard/docs"
```
I think we can have the `out/` directory in the `dashboard/` itself.

Samridhi-98 replied:
Done.

coditva commented on dashboard/package.json:
```diff
@@ -15,7 +15,8 @@
     "coverage:integration": "npm run test:integration && codecov",
-    "dev": "nodemon ./server/index.js"
+    "dev": "nodemon ./server/index.js",
+    "generate-docs": "jsdoc -c ../jsdoc.conf.json -d ../out/dashboard/docs"
```
Let's use the full name for the options. It'll make it easier to figure out what they mean without looking at the documentation.

We can also skip the `-d` option since we'll output to the `out/` directory only.
```suggestion
    "generate-docs": "jsdoc --configure ../jsdoc.conf.json"
```

Samridhi-98 replied:
Done.

coditva commented on package.json:
```diff
@@ -12,7 +12,8 @@
     "dev:client": "next ./frontend",
-    "build": "node npm/build.js"
+    "build": "node npm/build.js",
+    "generate-docs": "jsdoc -c jsdoc.conf.json -d ./out -r ./lib"
```
I think we can remove this since `lib/` is already taken care of by the individual directories/packages.

coditva commented on dashboard/package.json:
```diff
@@ -15,7 +15,8 @@
     "coverage:integration": "npm run test:integration && codecov",
-    "dev": "nodemon ./server/index.js"
+    "dev": "nodemon ./server/index.js",
+    "generate-docs": "jsdoc --configure ../jsdoc.conf.json"
```
We should maybe make this command similar to what we use in `newman` for consistency: `build-docs`

Ref: https://github.com/postmanlabs/newman/blob/develop/package.json#L33

coditva commented on jsdoc.conf.json:
```diff
@@ -0,0 +1,26 @@
+{
```
Let's also rename this file for consistency with `newman`: `.jsdoc-config.json`

Ref: https://github.com/postmanlabs/newman/blob/develop/.jsdoc-config.json

coditva commented on jsdoc.conf.json:
```diff
@@ -0,0 +1,26 @@
+        "cleverLinks": false,
+        "monospaceLinks": false
+    }
+}
```
Let's use the Postman's jsdoc theme and output options:
```json
"opts": {
    "template": "./node_modules/postman-jsdoc-theme",
    "encoding": "utf8",
    "destination": "./out/docs",
    "recurse": true,
    "readme": "README.md"
  },

  "markdown": {
    "parser": "gfm",
    "hardwrap": false
  }
```

Ref: https://github.com/postmanlabs/newman/blob/develop/.jsdoc-config.json

coditva commented on package.json:
```diff
@@ -12,7 +12,8 @@
     "dev:client": "next ./frontend",
-    "build": "node npm/build.js"
+    "build": "node npm/build.js",
+    "generate-docs": "jsdoc --configure jsdoc.conf.json -d ./out"
```
We can maybe remove the `generate-docs` command from this file itself since there are no source files? Another solution could be to generate docs for each of the child subdirectories.

See `test:unit` for example.

Samridhi-98 replied:
I am not getting this change will you please elaborate on it.
